prism_compile.c

```diff
@@ -1380,21 +1381,155 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
+                                            ADD_SEND(ret, &dummy_line_node, id_core_hash_merge_ptr, INT2FIX(cur_hash_size * 2 + 1));
+                                        }
+                                        else {
+                                            ADD_INSN1(ret, &dummy_line_node, newhash, INT2FIX(cur_hash_size * 2));
+                                            cur_hash_size = 0;
+                                            new_hash_emitted = true;
+                                        }
+                                    }
+
+                                    break;
+                                }
+                                case PM_ASSOC_SPLAT_NODE: {
+                                    if (len > 1) {
+                                        ADD_INSN1(ret, &dummy_line_node, putspecialobject, INT2FIX(VM_SPECIAL_OBJECT_VMCORE));
+                                        if (i == 0) {
+                                            ADD_INSN1(ret, &dummy_line_node, newhash, INT2FIX(0));
+                                            new_hash_emitted = true;
+                                        }
+                                        else {
+                                            PM_SWAP;
+                                        }
+                                    }
+
+                                    pm_assoc_splat_node_t *assoc_splat = (pm_assoc_splat_node_t *)cur_node;
+                                    PM_COMPILE(assoc_splat->value);
+
+                                    flags |= VM_CALL_KW_SPLAT | VM_CALL_KW_SPLAT_MUT;
+
+                                    ADD_SEND(ret, &dummy_line_node, id_core_hash_merge_kwd, INT2FIX(2));
+
+                                    if ((i < len - 1) && !PM_NODE_TYPE_P(keyword_arg->elements.nodes[i + 1], cur_type)) {
+                                        ADD_INSN1(ret, &dummy_line_node, putspecialobject, INT2FIX(VM_SPECIAL_OBJECT_VMCORE));
+                                        PM_SWAP;
+                                    }
+
+                                    cur_hash_size = 0;
+                                    break;
+                                }
+                                default: {
+                                    rb_bug("Unknown type");
+                                }
+                              }
+                          }
+                          break;
+                      }
+                      else {
+                          kw_arg = rb_xmalloc_mul_add(len, sizeof(VALUE), sizeof(struct rb_callinfo_kwarg));
+                          flags |= VM_CALL_KWARG;
+                          kw_arg->keyword_len += len;
+
+                          for (size_t i = 0; i < len; i++) {
+                              pm_assoc_node_t *assoc = (pm_assoc_node_t *)keyword_arg->elements.nodes[i];
+                              kw_arg->keywords[i] = pm_static_literal_value(assoc->key, scope_node, parser);
```

Comment:
I'm expecting this won't work for things like `foo(bar => baz)`. Can you add a comment here indicating that we'll need to revisit this?